File: java/java-sdk/src/main/java/io/kestra/sdk/model/Enforcement.java
/*
 * Kestra EE
 * All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
 */


package io.kestra.sdk.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Gets or Sets Enforcement
 */
public enum Enforcement {
  
  ACTIVE("ACTIVE"),
  
  EVALUATE("EVALUATE"),
  
  DISABLED("DISABLED"),
  
  REFERENCE("REFERENCE"),
  
  UNKNOWN_DEFAULT_OPEN_API("unknown_default_open_api");

  private String value;

  Enforcement(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static Enforcement fromValue(String value) {
    for (Enforcement b : Enforcement.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    return UNKNOWN_DEFAULT_OPEN_API;
  }
}

File: python/python-sdk/kestrapy/models/enforcement.py
# coding: utf-8

"""
    Kestra EE

    All API operations, except for Superadmin-only endpoints, require a tenant identifier in the HTTP path.<br/> Endpoints designated as Superadmin-only are not tenant-scoped.
"""  # noqa: E501


from __future__ import annotations
import json
from enum import Enum
from typing_extensions import Self


class Enforcement(str, Enum):
    """
    Enforcement
    """

    """
    allowed enum values
    """
    ACTIVE = 'ACTIVE'
    EVALUATE = 'EVALUATE'
    DISABLED = 'DISABLED'
    REFERENCE = 'REFERENCE'

    @classmethod
    def from_json(cls, json_str: str) -> Self:
        """Create an instance of Enforcement from a JSON string"""
        return cls(json.loads(json_str))
